src/dataviews/dataviews-factory.js

```diff
@@ -15,73 +15,59 @@ module.exports = Model.extend({
   initialize: function (attrs, opts) {
     if (!opts.map) throw new Error('map is required');
     if (!opts.vis) throw new Error('vis is required');
-    if (!opts.analysisCollection) throw new Error('analysisCollection is required');
     if (!opts.dataviewsCollection) throw new Error('dataviewsCollection is required');
-    if (!opts.analysisCollection) throw new Error('analysisCollection is required');
 
     this._map = opts.map;
     this._vis = opts.vis;
-    this._analysisCollection = opts.analysisCollection;
     this._dataviewsCollection = opts.dataviewsCollection;
-    this._analysisCollection = opts.analysisCollection;
   },
 
-  createCategoryModel: function (layerModel, attrs) {
-    _checkProperties(attrs, ['column']);
-    attrs = this._generateAttrsForDataview(layerModel, attrs, CategoryDataviewModel.ATTRS_NAMES);
+  createCategoryModel: function (attrs) {
+    _checkProperties(attrs, ['source', 'column']);
```

Comment:
Should we ensure the `source` attrs is an instance of an `AnalysisModel` here? We're doing so in the `LayersFactory` [now](https://github.com/CartoDB/cartodb.js/blob/layer-sources/src/vis/layers-factory.js#L99).

Reply:
I wouldn't because that check will be made by `dataview-model-base`, that it's called right here. Doing it in several places it's too much defensive, IMO.

Reply:
:+1: